sdk/tests/types/transaction_id.rs

```diff
@@ -12,13 +12,13 @@ use iota_sdk::types::{
 };
 use packable::PackableExt;
 
-const TRANSACTION_ID: &str = "0x52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649";
+const TRANSACTION_ID: &str = "0x52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c64900000000";
 
 #[test]
 fn debug_impl() {
     assert_eq!(
         format!("{:?}", TransactionId::from_str(TRANSACTION_ID).unwrap()),
-        "TransactionId(0x52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649)"
+        "TransactionId { hash: TransactionHash(0x52fdfc072182654f163f5f0f9a621d729566c74d10037c4d7bbb0407d1e2c649), slot_index: SlotIndex(0) }"
```

Comment:
I wouldn't do that, this is going to make our life way too hard by not being able to copy paste IDs to explorer/APIs

Reply:
this is only the debug print, not the display

Reply:
Yes I know

Reply:
yeah so I don't see a problem. it should give debug information

Reply:
The only pragmatic use is the full ID that we are going to copy paste everywhere. We've not done the split for OutputId for the exact same reason, we are not going to manually construct the ID to input it somewhere. If people want the slot index, they can log it themselves, but that's a very tiny usecase.

Reply:
I think it's more useful to have a debug impl that is actually functional for debugging. I can't currently read hex that easily so if I need to know the slot index I have to use a third party tool

Reply:
If you need to know the slot index, print the slot index? What third party tool? The slot index being part of the ID really is just an internal thing, no one will care about that.

Reply:
We can't degrade the UX that much for a just tiny use case 🤷🏻

Reply:
I'm saying if we print the debug representation it prints hex, and I would manually have to print the slot index because there's no good debugging representation of the type. We're not degrading the UX, this is making it *better*. If you want the hex ID, the display impl exists for that purpose.

Reply:
Sorry but I disagree. This definitely makes UX worse. You can't know how things will be printed, they could be nested in other things that are preferable as Debug or simply don't have Display. It was never done for OutputId, nobody complained because the output index is simply not something people are interested in, same for slot index which is arguably even less interesting than output index. People would 100% have complained if OutputId Debug was printing the two parts.